solr/core/src/java/org/apache/solr/client/solrj/embedded/JettySolrRunner.java

```diff
@@ -611,27 +608,11 @@ public void stop() throws Exception {
     Map<String,String> prevContext = MDC.getCopyOfContextMap();
     MDC.clear();
     try {
-      Filter filter = dispatchFilter.getFilter();
-
       // we want to shutdown outside of jetty cutting us off
-      SolrDispatchFilter sdf = getSolrDispatchFilter();
-      ExecutorService customThreadPool = null;
-      if (sdf != null) {
-        customThreadPool = ExecutorUtil.newMDCAwareCachedThreadPool(new SolrNamedThreadFactory("jettyShutDown"));
-
-        sdf.closeOnDestroy(false);
-//        customThreadPool.submit(() -> {
-//          try {
-//            sdf.close();
-//          } catch (Throwable t) {
-//            log.error("Error shutting down Solr", t);
-//          }
-//        });
-        try {
-          sdf.close();
-        } catch (Throwable t) {
-          log.error("Error shutting down Solr", t);
-        }
+      try {
```

Comment:
This change is fine in general, but it has test and production shutdown ramifications that I can't reason about mentally.

For tests, if the JettySolrRunner stop timeout is set to 0, it may not change much, if I recall correctly, Jetty may not interrupt its executor threads unless it's greater than 0. This was originally added to avoid those interrupts. when the stop timeout is > 0. I wouldn't recommend that behavior, but alterations do have a wide range of test and production stop affects.

If the production jetty configuration is still set to a 5 second stop timeout, that means jetty will start interrupting its threads at 2.5 seconds, and. with this change that. means interrupting Solr shutdown, which would now be on a jetty pool thread. This was likely done because Solr behaved poorly in tests in this case. So current ramifications unknown. Ideal case is that Solr is fine and graceful in production and tests being interrupted on shutdown though.

Reply:
Next time please add more code comments to justify the complexity you added so that mortals like myself know why it's there.

A background close thread was added 11/29/2018 but you commented it out 2 days later, leaving `customThreadPool` and some commented code even though it's not used anymore.  Ugh.  So I'm unsure how to reconcile the current state with your explanation above.  You explain why a background thread is needed to close (interesting) but there isn't one today and hasn't been there for years.  🤷